### ISSUE TITLE
"Could not download Sanesecurity public GPG key"

### ISSUE THREAD

**Issue by plskuser** [extremeshok/clamav-unofficial-sigs#316]
Hello,

/usr/sbin/clamav-unofficial-sigs.sh send cron mails like "Could not download Sanesecurity public GPG key".

The script fails:
```
# /usr/sbin/clamav-unofficial-sigs.sh
################################################################################
 eXtremeSHOK.com ClamAV Unofficial Signature Updater
 Version: v7.0.1 (2020-01-25)
 Required Configuration Version: v91
 Copyright (c) Adrian Jon Kriel :: admin@extremeshok.com
################################################################################
Loading config: /etc/clamav-unofficial-sigs/master.conf
Loading config: /etc/clamav-unofficial-sigs/os.conf
Loading config: /etc/clamav-unofficial-sigs/user.conf
===================
Preparing Databases
===================
*****************************************************
ALERT: Could not download Sanesecurity public GPG key
*****************************************************
```

curl request  to http://sanesecurity.com/publickey.gpg shows "403 Forbidden" from server " BitNinja Captcha Server"

```
# curl -vvv -I  http://sanesecurity.com/publickey.gpg
* About to connect() to sanesecurity.com port 80 (#0)
*   Trying 185.139.129.91...
* Connected to sanesecurity.com (185.139.129.91) port 80 (#0)
> HEAD /publickey.gpg HTTP/1.1
> User-Agent: curl/7.29.0
> Host: sanesecurity.com
> Accept: */*
>
< HTTP/1.1 403 Forbidden
HTTP/1.1 403 Forbidden
< Connection: close
Connection: close
< Cache-Control: no-cache, no-store, must-revalidate
Cache-Control: no-cache, no-store, must-revalidate
< Pragma: no-cache
Pragma: no-cache
< Expires: 0
Expires: 0
< Server: BitNinja Captcha Server
Server: BitNinja Captcha Server
< Content-Length: 9306
Content-Length: 9306
```

As  I can see the URL is defined in "/etc/clamav-unofficial-sigs/master.conf"

```
# grep  sanesecurity_gpg_url /etc/clamav-unofficial-sigs/master.conf
sanesecurity_gpg_url="http://www.sanesecurity.net/publickey.gpg"

# grep  sanesecurity_gpg_url /usr/sbin/clamav-unofficial-sigs.sh
    xshok_file_download "${work_dir_gpg}/publickey.gpg" "$sanesecurity_gpg_url"
```

Is there any blacklist on your side? 
From another server the curl is successful.

**Comment by sirthias** [comment]
I just ran into a very similar problem:
My VPS (Hetzner Cloud) cannot even connect to `sanesecurity.com` port 80, even though trying the same from other boxes or my dev machine works fine.
I was able to work around the problem by downloading the `publickey.gpg` manually from another box and putting it into some alternative location and then change the `sanesecurity_gpg_url` in `/etc/clamav-unofficial-sigs/master.conf` accordingly.
With this patch `/usr/sbin/clamav-unofficial-sigs.sh` runs to completion successfully.
